### ISSUE TITLE
font icons showing 4-digit garble instead 

### ISSUE THREAD

**Issue by 3cstudio** [virtualmin/virtualmin-gpl#57]
I seem to be having FontAwesome issues in Firefox 60.0 for Mac. 

I see a few references to this issue online, but not for FF. Do you have any past / known issues along this line? I can see no Preference to properly refer to the FontAwesome icon fonts.

Please advise -- much appreciated!
<img width="397" alt="screen shot 2018-05-15 at 3 42 18 pm" src="https://user-images.githubusercontent.com/942412/40080348-a94e83da-5858-11e8-9717-de5d04a73ad0.png">
<img width="1642" alt="screen shot 2018-05-15 at 10 20 39 am" src="https://user-images.githubusercontent.com/942412/40080350-a960cfc2-5858-11e8-862f-f5b3d1f0d006.png">

Steve

**Comment by swelljoe** [comment]
Check the language settings in Webmin (and your browser). Any non-Unicode character set would result in fonticons not working right, I think. There are a few places it can be wrong...there's a "Character set for HTML pages" option in the Webmin config, as well as one that sets it to use the language specified by the browser. I think either of those could trigger non-Unicode.

This is just a guess...but, I seem to recall this being a thing that could happen.

I think it could also happen if the icon font can't be loaded...maybe? Again, guessing. I've never seen this specific behavior. Ilia might have more insight.

**Comment by 3cstudio** [comment]
Thanks for the response, Joe. Attached is what I see in: Webmin > Usermin Configuration . . . if that was what you're referring to? I don't see anything related to "Character sets" here, and I'm unsure of where in the myriad other options to seek out the potential option for their being an errant reference to non-Unicode characters.
Thanks for sticking with me on this odd issue -- small but rather annoying. :)
Steve
<img width="1639" alt="screen shot 2018-05-15 at 4 44 08 pm" src="https://user-images.githubusercontent.com/942412/40082697-c0041c14-585f-11e8-83b9-d53834603259.png">

**Comment by qooob** [comment]
What is shown in Safari or Chrome?

Lanuage and charset in _Webmin/Webmin Configuration/Language and Locale_ or in _Webmin/Change Language and Theme_. Take a heed, that latter module has priority and overwrites all settings set globally in the first module, as meant to work as user based.

There are quite a few issues with `font awesome displaying squares` if you search for it. The only guess right now, besides fancy settings in mentioned modules, could be that _woof2_ is not supported on Firefox Mac browser (which I doubt; besides we're not using _woof2_ but _woof_ unless I accidentally embedded newer format - I'll double check just a bit later).

**Comment by 3cstudio** [comment]
Thanks so much, Ilia.

I have colleagues who login via Firefox for Mac and it performs as it should. I login via Chrome and things are proper there as well. So it appears to be isolated to my machine's copy of Firefox... I have an older MacBook Pro, so I logged in on that machine -- it too loads all icons properly, even after it updated from 59.0x to 60.0 upon checking the About page. I am going to try loading Firefox brand new on this newer machine and see if that works or not.

Thanks again, I will report back with what I find out.

Steve

**Comment by 3cstudio** [comment]
Thanks so much, Ilia.

I have colleagues who login via Firefox for Mac and it performs as it should. I login via Chrome and things are proper there as well. So it appears to be isolated to my machine's copy of Firefox... I have an older MacBook Pro, so I logged in on that machine -- it too loads all icons properly, even after it updated from 59.0x to 60.0 upon checking the About page. I am going to try loading Firefox brand new on this newer machine and see if that works or not.

Thanks again, I will report back with what I find out.
UPDATE: I ended up loading (overwriting) FF and opening Virtualmin again after logging in. The icons were still missing, so I went into the Add-ons Manager and examined Extensions once more -- the only difference between my old machine and this newer one was the newer one had an extension called "Offers.com Coupon Codes and Deals" (which I don't remember adding??). I disabled it -- the icons magically reappeared(!). Enabling it once again made them defective once again. I will be deleting it entirely, as it is rather suspicious in the first place!

This seems to be the "fix". Thank you for sticking with me on this, and hopefully this may help someone else with this odd problem in the future.

Best,

Steve C.
3Cstudio

Steve
